### DISCUSSION TITLE
Server installation in text mode

### DISCUSSION THREAD

**Discussion by Plekhanov-sgu** [ltsp/ltsp#190] [Q&A]
In the old version of LTSP on the server, for example, I run: "ltsp-build-client --dist xenial --arch i386 --fat-client-desktop mate-desktop"
Gathered in: / opt / ltsp desired configuration
Then using: "ltsp-chroot"
I installed the necessary packages in text mode.
Having quit chroot, I collected from / opt / ltsp / nbd-image for clients using: "ltsp-update-image"
The new version has been tested on a desktop computer and works great.
 I got confused in the descriptions of whether it is possible to collect the required configuration on the server in text mode (unfortunately, the server resource hardly supports desktop installation).

**Comment by alkisg** [comment]
The new LTSP version supports chroots and text mode servers. But it doesn't have the ltsp-build-client and ltsp-chroot tools.

Instead of ltsp-build-client, you can use the `debootstrap` command, or alternatives like `systemd-nspawn`, lxc etc. Instead of ltsp-chroot, you can use the `chroot` command. And even qemu-kvm supports both chroots and VMs in text mode, so one can boot and maintain a chroot in text mode.

The reason for not including these tools in the new LTSP is that you can't really explain or support them in a couple of wiki pages. If sysadmins need to maintain chroots, they should google and read about them, and then they should be able to maintain them fine without the little ltsp-build-client or ltsp-chroot scripts.

It's the same for VM images too. We won't have a lot of wiki pages on how to create VMs using virt-manager or VirtualBox; one can google and learn how to create VMs. LTSP can then use chroots, VMs, or the server / (chrootless) as the source.

Another idea is to export /srv/ltsp via NFS read-write mode to a desktop client, and manage VMs or chroots graphically from the desktop client, even if the NFS share is on the server.

**Comment by Plekhanov-sgu** [comment]
Thanks to Alkis for the answer, you are absolutely right in the ideology of using the server desktop, it takes an order of magnitude less time to install applications for users. I tried using debootstrap and alternatives, it’s easier to install a second network card by assigning it the address 192.168. 67.1 (without external DHCP). A guest account has been created for customers to log in. It is not possible to save files on clients after exiting a session. How to synchronize folders on the server and clients? This is necessary to distribute tasks to students and save their calculation results, for example.

**Comment by alkisg** [comment]
If your "guest accounts" have different data, you should have multiple guest accounts.
pc01 should autologin with guest01,
pc02 should autologin with guest02, etc.

Then you can do whatever you want with /home/guestXX server-side.

**Comment by Plekhanov-sgu** [comment]
Thanks Alkis. What could be the reason if guestXX saves the file in its / home / "existing folder", but this file is not saved on the server.

**Comment by alkisg** [comment]
That sounds strange; login as guestXX and paste the output of these commands:

```shell
grep home /proc/self/mountinfo
ls -l /home
df -h
```

**Comment by Plekhanov-sgu** [comment]
Тhank you,I can only do this tomorrow, and by placing the file in the / home / guestXX folder on the server side, it should appear on the client side, I understand correctly?

**Comment by alkisg** [comment]
Yes. In the worse case, you'll have to press F5 on your file manager to see it.

**Comment by Plekhanov-sgu** [comment]
terminal@ltsp53:~$ grep home /proc/self/mountinfo
terminal@ltsp53:~$ ls -l /home
итого 0 (total 0)
drwxr-xr-x 15 terminal terminal 380 окт 18 11:30 terminal
terminal@ltsp53:~$ df -h
Файл.система   Размер Использовано  Дост Использовано% Cмонтировано в
(File System Size Used Access Used% Mounted in)
udev             874M            0  874M            0% /dev
tmpfs            182M         2,8M  179M            2% /run
tmpfs            908M          37M  872M            4% /run/initramfs/ltsp
overlay          908M          37M  872M            4% /
tmpfs            908M          15M  894M            2% /dev/shm
tmpfs            5,0M         4,0K  5,0M            1% /run/lock
tmpfs            908M            0  908M            0% /sys/fs/cgroup
tmpfs            182M          52K  182M            1% /run/user/1001

**Comment by alkisg** [comment]
I think you run these before the user has logged in...
Can you run them again after login?

**Comment by Plekhanov-sgu** [comment]
This is after logging in, looked at the network address of the server 192.168.0.1, .Alkis, the address must be 192.168.67.1? It is necessary?
Epoptes sees the terminal, and the process running in it. Epoptes performs any action on
client

**Comment by alkisg** [comment]
Can the terminal user login on the server?
Maybe you didn't create a home for him?
What's the output of the following command on the server?

```shell
getent passwd terminal
```

**Comment by alkisg** [comment]
The server doesn't need 2 network cards and the 192.168.67.1 IP is only recommended when you have an isolated, separate LTSP subnet.

I.e. most people don't use that.

**Comment by Plekhanov-sgu** [comment]
I have a separate LTSC subnet, I can’t do without it, this is a subnet in the large network of our university, I reconfigured the second card to the address 192.168.67.1

**Comment by Plekhanov-sgu** [comment]
Sometimes I do not see the obvious. The user was without a password when entering the terminal.
